### ISSUE TITLE
I  think this a bug. caption.py    140

### ISSUE THREAD

**Issue by tigerbrother222** [sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning#196]
incomplete_inds = [ind for ind, next_word in enumerate(next_word_inds) if  next_word != word_map['<end>']]
incomplete_inds always is [0,1,2,3,4] .   
and then 
complete_inds = list(set(range(len(next_word_inds))) - set(incomplete_inds))
complete_inds  is empty
so complete_seqs is null.

my Model out is error

**Comment by ArborealSky** [comment]
I have the same bug in caption.py . Did you fix it ?

**Comment by ArborealSky** [comment]
i fix it , you should consider the situation that your model is too weak to generate the `<end>` symbol for ending your sentences.

**Comment by ArborealSky** [comment]
please add code 
```
if len(complete_seqs_scores) == 0:
        return seqs[0], 0
```
before

```
i = complete_seqs_scores.index(max(complete_seqs_scores))
```
in fuction `caption_image_beam_search`

**Comment by gsoykan** [comment]
why don't you use it like the following ? Is there any specific reason why you are returning 0 instead of matching attention map? 
```
if len(complete_seqs_scores) == 0:
    return seqs[0], seqs_alpha[0]
```

**Comment by ArborealSky** [comment]
You are right. Actually, the code in my project doesn't need to care about seqs_alpha. So I just used '0' to replace its return. Your code are more complete and better.


王佳旭
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月4日(星期四) 凌晨0:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning] I  think this a bug. caption.py    140 (Issue #196)





 
why don't you use it like the following ? Is there any specific reason why you are returning 0 instead of matching attention map?
 if len(complete_seqs_scores) == 0:     return seqs[0], seqs_alpha[0]

**Comment by ArborealSky** [comment]
> why don't you use it like the following ? Is there any specific reason why you are returning 0 instead of matching attention map?
> 
> ```
> if len(complete_seqs_scores) == 0:
>     return seqs[0], seqs_alpha[0]
> ```

You are right. Actually, the code in my project doesn't need to care about seqs_alpha. So I just used '0' to replace its return. Your code are more complete and better.
